ckb-lock-common/src/simple_cursor.rs

```diff
@@ -0,0 +1,84 @@
+use core::fmt::Display;
+
+use crate::error::Error;
+use alloc::{boxed::Box, vec::Vec};
+use ckb_std::{
+    ckb_constants::Source,
+    syscalls::{load_witness, SysError},
+};
+use molecule2::{Cursor, Read};
+
+pub struct SimpleCursor {
+    pub offset: u32,
+    pub size: u32,
+}
+
+impl SimpleCursor {
+    pub fn new(offset: u32, size: u32) -> Self {
+        Self { offset, size }
+    }
+    pub fn new_from_cursor(cursor: &Cursor) -> Self {
+        Self {
+            offset: cursor.offset as u32,
+            size: cursor.size as u32,
+        }
+    }
+
+    pub fn parse(str: &str) -> Result<Self, Error> {
+        let result: Vec<u32> = str
+            .split_terminator(":")
+            .into_iter()
+            .map(|n| u32::from_str_radix(n, 16).unwrap())
```

Comment:
.unwrap() => .ok_or(Error::WrongHex)?

Reply:
`Result` implements `FromIterator`, if an `Result::Err` is found, the iteration will terminate and return early:

```rust
        let result: Result<Vec<_>, _> = str
            .split_terminator(":")
            .into_iter()
            .map(|n| u32::from_str_radix(n, 16))
            .collect();
```

Reply:
Thanks for sharing. Currently, it panics immediately: it is what we need.

Reply:
this is replied to @mohanson suggestion, we can use `.ok_or(Error::WrongHex)?` on the collected `Result` instead of panics.